### ISSUE TITLE
Error building docker and docker-slim images

### ISSUE THREAD

**Issue by danielpacak** [aquasecurity/tracee#1165]
I'm trying to build docker and docker-slim images with no success. Both `make docker` and `make docker-slim` targets fail with the following error:

```console
$ make docker-slim
[ ... truncated output ... ]
clang \
	-I dist/libbpf/usr/include/bpf \
	-I 3rdparty/include \
	-I tracee/x86_64 \
	-DCORE \
	-D__TARGET_ARCH_x86 \
	-D__BPF_TRACING__ \
	-D__VM_LINUX_H__ \
	-D__KERNEL__ \
	-DLINUX_VERSION_CODE=328704 \
	-DRHEL_RELEASE_CODE=0 \
	-target bpf \
	-O2 -c -g -o dist/tracee.bpf.core.o tracee/tracee.bpf.c
tracee/tracee.bpf.c:774:9: error: use of unknown builtin '__builtin_preserve_type_info' [-Wimplicit-function-declaration]
    if (bpf_core_type_exists(union kernfs_node_id)) {
        ^
dist/libbpf/usr/include/bpf/bpf_core_read.h:155:2: note: expanded from macro 'bpf_core_type_exists'
        __builtin_preserve_type_info(*(typeof(type) *)0, BPF_TYPE_EXISTS)
        ^
tracee/tracee.bpf.c:774:9: note: did you mean '__builtin_preserve_access_index'?
dist/libbpf/usr/include/bpf/bpf_core_read.h:155:2: note: expanded from macro 'bpf_core_type_exists'
        __builtin_preserve_type_info(*(typeof(type) *)0, BPF_TYPE_EXISTS)
        ^
tracee/tracee.bpf.c:589:35: note: '__builtin_preserve_access_index' declared here
    struct mnt_namespace* mntns = READ_KERN(ns->mnt_ns);
                                  ^
tracee/tracee.bpf.c:329:27: note: expanded from macro 'READ_KERN'
                          bpf_core_read((void *)&_val, sizeof(_val), &ptr);             \
                          ^
dist/libbpf/usr/include/bpf/bpf_core_read.h:206:47: note: expanded from macro 'bpf_core_read'
        bpf_probe_read_kernel(dst, sz, (const void *)__builtin_preserve_access_index(src))
                                                     ^
tracee/tracee.bpf.c:774:9: warning: indirection of non-volatile null pointer will be deleted, not trap [-Wnull-dereference]
    if (bpf_core_type_exists(union kernfs_node_id)) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dist/libbpf/usr/include/bpf/bpf_core_read.h:155:31: note: expanded from macro 'bpf_core_type_exists'
        __builtin_preserve_type_info(*(typeof(type) *)0, BPF_TYPE_EXISTS)
                                     ^~~~~~~~~~~~~~~~~~
tracee/tracee.bpf.c:774:9: note: consider using __builtin_trap() or qualifying pointer with 'volatile'
dist/libbpf/usr/include/bpf/bpf_core_read.h:155:31: note: expanded from macro 'bpf_core_type_exists'
        __builtin_preserve_type_info(*(typeof(type) *)0, BPF_TYPE_EXISTS)
                                     ^
1 warning and 1 error generated.
make[1]: *** [Makefile:134: dist/tracee.bpf.core.o] Error 1
make[1]: Leaving directory '/tracee/tracee-ebpf'
make: *** [Makefile:44: dist/tracee-ebpf] Error 2
The command '/bin/sh -c make' returned a non-zero code: 2
make: *** [Makefile:67: docker-slim] Error 2
```

Git revision: 9962191d9554237e5138fd7f051316e3e9389de7  
Env: Vagrant in this repo (vagrant up && vagrant ssh). With `sudo apt install docker.io` installed.

**Comment by danielpacak** [comment]
As we discussed with @rafaeldtinoco the build failure is related to the pending changes in #1160

**Comment by rafaeldtinoco** [comment]
and the ones done by https://github.com/aquasecurity/tracee/pull/1130. discussed at: https://github.com/aquasecurity/tracee/pull/1130#pullrequestreview-812809549

**Comment by yanivagman** [comment]
> As we discussed with @rafaeldtinoco the build failure is related to the pending changes in #1160 

I'm not sure #1160 will resolve the error you face.
I think it is also related to the fact that we are using an old clang version, where __builtin_preserve_type_info builtin didn't exist

**Comment by danielpacak** [comment]
> > As we discussed with @rafaeldtinoco the build failure is related to the pending changes in #1160
> 
> I'm not sure #1160 will resolve the error you face here. I think it is also related to the fact that we are using an old clang version, where __builtin_preserve_type_info builtin didn't exist

Actually I created this issue because bumping up Alpine to 3.15 to install Clang 12 to resolve #1156 did not help.

**Comment by yanivagman** [comment]
@danielpacak using the current Makefile I can see that the clang version used is 11 and not 12.
Did you change the Dockerfile locally?

**Comment by danielpacak** [comment]
> @danielpacak using the current Makefile I can see that the clang version used is 11 and not 12.
> Did you change the Dockerfile locally?

Yes, I changed Dockerfile locally to see if Clang 12 is in the main repo of Alpine 3.15

**Comment by yanivagman** [comment]
The clang package used by golang:1.16-alpine (our builder image) is 11 and not 12.
Did you use the alpine image instead?

**Comment by danielpacak** [comment]
> The clang package used by golang:1.16-alpine (our builder image) is 11 and not 12.
> Did you use the alpine image instead (of golang:1.16-alpine)?

I see what you mean now. There are slim and fat variants built on alpine and I missed `golang:1.16-alpine` which indeed seems to be based on 3.14.

Since there is no `golang:1.16-alpine3.15` available from the DockerHub we can use alpine:3.15 as a base image of the builder and then install go package on it. I actually tried it and I no longer have the compilation error!
